src/Rules/RealEmail.php

```diff
@@ -65,19 +65,31 @@ public function passes($attribute, $email): bool
             }
         }
 
-        $host = (string) substr($email, strrpos($email, '@') + 1);
-
-        if ($this->hasConstraint('mx') && ! $this->checkMX($host)) {
-            $this->messages[] = __('realEmailValidation::messages.mx');
+        $hostSeparatorPosition = strrpos($email, '@');
+        if ($hostSeparatorPosition === false || $hostSeparatorPosition === strlen($email) - 1) {
+            $this->messages[] = __('realEmailValidation::messages.nohost');
             if ($this->bail) {
                 return false;
             }
-        }
+        } else {
+            $emailTail = (string) substr($email, $hostSeparatorPosition + 1);
+            $host = idn_to_ascii($emailTail, 0, INTL_IDNA_VARIANT_UTS46);
```

Comment:
`idn_to_ascii` requires `ext-intl` or a polyfill (eg `symfony/polyfill-intl-idn`). I think it's better to add `symfony/polyfill-intl-idn` as dependency

Reply:
Done.

The Scrutinizer, PHP Analyzer test fails and recommends commiting the composer.lock, but there seems to be divided opinions about doing this. My understanding is that this is not recommended for libraries.

The Travis PHP 7.3 build fails, my research leads me to believe this is from a bug in PHP 7.3, citing [https://groups.google.com/forum/#!topic/linux.debian.bugs.dist/SEkt7J1kgW4](https://groups.google.com/forum/#!topic/linux.debian.bugs.dist/SEkt7J1kgW4)

Reply:
thanks!